site/app/templates/forum/ShowCategories.twig

```diff
@@ -67,33 +72,40 @@
                 reorderCategories('{{ csrf_token }}');
             }
         });
-        $("#ui-category-list").find(".fa-trash").click(function() {
-            var item = $(this).parent().parent().parent();
-            var category_id = parseInt(item.attr('id').split("-")[1]);
-            var category_desc = item.find(".categorylistitem-desc span").text().trim();
+
+        function getItem(input) {
+            return $(input).closest('.category-item-box');
+        }
+        
+        $(".delete-category-button").click(function() {
+            const item = getItem(this);
+            const category_id = item.data('category_id');
+            const category_desc = item.data('category_desc');
             deleteCategory(category_id, category_desc, '{{ csrf_token }}');
         });
-        $("#ui-category-list").find(".fa-edit").click(function() {
-            var item = $(this).parent().parent().parent();
-            var category_desc = item.find(".categorylistitem-desc span").text().trim();
+
+        $(".edit-category-desc-button").click(function() {
+            const item = getItem(this);
+            const category_desc = item.data('category_desc');
+            console.log(item, category_desc)
             item.find(".categorylistitem-editdesc input").val(category_desc);
             item.find(".categorylistitem-desc").hide();
             item.find(".categorylistitem-editdesc").show();
 
         });
-        $("#ui-category-list").find(".fa-times").click(function() {
-            var item = $(this).parent().parent().parent();
+        $(".cancel-desc-button").click(function() {
+            const item = getItem(this);
             item.find(".categorylistitem-editdesc").hide();
             item.find(".categorylistitem-desc").show();
         });
 
-        $("#ui-category-list").find(".fa-check").click(function() {
-            var item = $(this).parent().parent().parent();
-            var category_id = parseInt(item.attr('id').split("-")[1]);
-            var category_desc_original = item.find(".categorylistitem-desc span").text().trim();
-            var category_desc = item.find("input").val().trim();
+        $(".save-desc-button").click(function() {
+            const item = getItem(this);
+            const category_id = item.data('category_id');
+            const category_desc_original = item.data('category_desc');//item.find(".categorylistitem-desc span").text().trim();
```

Comment:
Should we remove that extra commented code?

Reply:
Ah yes, sorry I forgot to clean up once I was done.